### PR TITLE
fixes esm exports

### PR DESCRIPTION
**Pull request by jaredperreault-okta**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
       },
       "browser": {
         "types": "./build/lib/index.d.ts",
-        "import": "./build/esm/browser/index.js"
+        "import": "./build/esm/browser/index.js",
+        "default": "./build/esm/browser/index.js"
```
Comment (jaredperreault-okta) [inline package.json:22]: `jest@28` updated it's resolver algorithm, jest will now resolve the `"default"` export

File: scripts/build.js [hunk 2]
```diff
@@ -98,7 +98,10 @@ fs.writeFileSync(`${BUILD_DIR}/package.json`, JSON.stringify(packageJSON, null,
 
 // Add an extra package.json underneath ESM to indicate module type
 // This helps tools like Jest identify this code as ESM
+// (name and version are required for MFE)
```
Comment (jaredperreault-okta) [inline scripts/build.js:101]: This version of auth-js fails Module Federation builds (via webpack 5). Seems related to https://github.com/webpack/webpack/issues/13457. Adding `name` and `version` to this "dummy" `package.json` fixes the issue

File: package.json (hunk 1 above)
Comment (shuowu-okta) [reply package.json:22]: jest@28 starts using the esm bundle by default? or config will be needed?

Comment (jaredperreault-okta) [reply package.json:22]: I'm not sure it prioritizes esm bundle specifically (jest considers it's esm support to be experimental), but it changed the preference of which exports it consumes.
(I turned off the `moduleMapper` config when testing this)

More info here: https://github.com/okta/okta-auth-js/issues/1268

**Review by shuowu-okta — APPROVED** [review]
please also add changelog

**Pull request merged**